Services/AspireService.cs

```diff
@@ -1,93 +1,146 @@
             }
-            Console.WriteLine($"WatchResources connection canceled");
-        }
-
-        public override async Task WatchResourceConsoleLogs(WatchResourceConsoleLogsRequest request, IServerStreamWriter<WatchResourceConsoleLogsUpdate> responseStream, ServerCallContext context)
-        {
-           var podName = request.ResourceId.Uid.ToString();
-           while(!context.CancellationToken.IsCancellationRequested)
-            {
-                using k8s.Autorest.HttpOperationResponse<Stream> stream = await _kubernetesClient.CoreV1.ReadNamespacedPodLogWithHttpMessagesAsync(podName, _kubernetesConfig.Namespace);
-                var logsUpdate = new WatchResourceConsoleLogsUpdate();
-                using var reader = new StreamReader(stream.Body);
-                while (!reader.EndOfStream)
-                {
-                    var logEntry = await reader.ReadLineAsync();
-                    logsUpdate.LogLines.Add(new ConsoleLogLine { Text = logEntry });
-                }
-                await responseStream.WriteAsync(logsUpdate);
-                // Introduce delay for testing
-                await Task.Delay(1000);
-            }
-        }
-
-        public override Task<ResourceCommandResponse> ExecuteResourceCommand(ResourceCommandRequest request, ServerCallContext context)
-        {
-            return base.ExecuteResourceCommand(request, context);
-        }
-
-    }
-}
+            Console.WriteLine($"WatchResources connection canceled");
+        }
+
+        private async Task WriteInitialDataToStream(IServerStreamWriter<WatchResourcesUpdate> responseStream)
+        {
+            // Gets the initial data and return it
+
+            var podsList = await _kubernetesClient.CoreV1.ListNamespacedPodAsync(_kubernetesConfig.Namespace);
+            foreach (var pod in podsList)
+            {
+                var labels = pod.Labels();
+                if (labels.IsNullOrEmpty())
+                {
+                    _logger.LogDebug($"Unable to fetch app name because Pod: {pod.Metadata.Name} does not contain labels.Skipping to next pod.");
+                    continue;
+                }
+
+                var result = labels.TryGetValue("containerapps.io/app-name", out string appName);
+                if (!result)
+                {
+                    _logger.LogDebug($"The Pod {pod.Metadata.Name} does not have an entry for the key app. Skipping to next pod.");
+                    continue;
+                }
+
+                _logger.LogDebug($"App name: {appName}");
+                _logger.LogDebug($"Result: {result}");
+                var initialData = new WatchResourcesUpdate()
+                {
+                    InitialData = new InitialResourceData()
+                    {
+                        Resources = { new Resource() { Name = appName, Commands = { new ResourceCommandRequest() { CommandType = "Restart" } } } }

```

Comment:
It looks like we're not sending the pods' data in this initialData. Any reasons not to add it?